### PR TITLE
V2 Linter API support

### LINKED ISSUE #3: Upcoming linter changes
We're in the process of replacing `linter` with a rewrite. Our central issue for tracking the plan is https://github.com/AtomLinter/linter-plus/issues/29. We're working hard to make this a smooth transition for you and for your users.

Please let us know here if you'd prefer to do the work yourself. If you have any questions or concerns please join us in our [room on gitter](https://gitter.im/AtomLinter/Linter).

### LINKED ISSUE #4: Implement specs
It looks like this linter does not currently have any specs written for it, adding it to the list being maintained [here](https://github.com/AtomLinter/Meta/issues/11) for tracking.

Specs do not need to be super detailed, as they should mainly test whether the package is properly running the linter plugin and parsing its messages. Leave testing of the linter functionality to the author of the linter.

If you need any help feel free to ask!

### LINKED ISSUE #8: Support v2 Linter API
In the [v2.3.0 release](https://github.com/steelbrain/linter/releases/tag/v2.3.0) of `linter` support for the v1 API was removed, breaking semver. This means that this provider is no longer supported by the current version of `linter`.

To get this provider working again it will need to be upgrade to the v2 API, a guide for accomplishing that can be found here: [Upgrading to Standard Linter v2](https://github.com/steelbrain/linter/blob/v2.3.0/docs/guides/upgrading-to-standard-linter-v2.md).

Until this is fixed in order to use this provider you will need to downgrade to the last version of `linter` that supports the v1 API, you can do this with `apm install linter@2.2.0`.

### PR DESCRIPTION
**Pull request by vzamanillo**
Fixes #3.
Fixes #4.
Fixes #8.

Plus CI.

### CODE REVIEW

File: lib/init.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        const fileText = editor.getText();
+        loadDeps();
+
+        const messages = [];
+        try {
+          bailey.parseString(fileText);
+        } catch (e) {
+          if (e instanceof bailey.ParserError) {
+            messages.push({
+              severity: levelMapping[e.inner.name],
```
Comment (Arcanemagus) [inline lib/init.js]: What happens if it's not one of the three mappings you have defined? The `Set` or other approaches you have used elsewhere should work fine here.

File: lib/init.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      if (atom.inDevMode()) {
+        // eslint-disable-next-line no-console
+        console.log('linter-bailey: All dependencies installed.');
+      }
+    };
+    depsCallbackID = window.requestIdleCallback(installLinterBaileyDeps);
+    this.idleCallbacks.add(depsCallbackID);
+
+    if (atom.inDevMode()) {
+      /* eslint-disable no-console */
```
Comment (Arcanemagus) [inline lib/init.js]:
```suggestion
      // eslint-disable-next-line no-console
```

File: lib/init.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      console.log('linter-bailey: bailey linter is now activated.');
+      /* eslint-enable no-console */
```
Comment (Arcanemagus) [inline lib/init.js]:
```suggestion
```

**Review by Arcanemagus — CHANGES_REQUESTED** [review]
I've enabled CircleCI here as well, so the next commit should trigger a build.

**Review by Arcanemagus — APPROVED** [review]
Thanks!

**Comment by Arcanemagus** [comment]
Hmmm, looks like something happened with the beta branch of Atom to break the specs.

**Comment by vzamanillo** [comment]
> Hmmm, looks like something happened with the beta branch of Atom to break the specs.

Yep, `Failed to load package 'language-bailey'`

**Comment by Arcanemagus** [comment]
Whoops, forgot to comment back here:

Filed as https://github.com/atom/atom/issues/18983 and fixed in https://github.com/atom/atom/pull/18984, since it's breaking existing packages it should be included in the next beta release 😉.

**Pull request merged**